[airq] Air-Q binding Initial contribution (#10048)
openhab/openhab-addons: bundles/org.openhab.binding.airq/README.md

# air-Q Binding

> The air-Q Binding integrates the air analyzer <a href="http://www.air-q.com">air-Q</a> device into the openHAB system.

Reviewer: Markdown syntax is preferred over HTML in this document, so that you can read it also unrendered. Same for the images.
```suggestion
The air-Q Binding integrates the air analyzer [air-Q](http://www.air-q.com) device into the openHAB system.
```

With the binding, it is possible to subscribe to all data delivered by the air-Q device.

> <img src="src/main/resources/image_air-Q.png" alt="air-Q image" width="400px" height="324px" />

Reviewer: Readme assets should be located in the `doc/` folder in the root of this binding.

## Supported Things

Only one Thing is supported: The `air-Q` device.
This Binding was tested with an `air-Q Pro` device with 14 sensors. It also works with an `air-Q` device with 11 sensors.

## Discovery

Auto-discovery is not yet supported.

## Binding Configuration

> The binding does not need to be configured.

Reviewer: Then, you could remove this.

## Thing Configuration

The air-Q Thing must be configured with (both mandatory):
| Parameter | Description                        |
|-----------|------------------------------------|
| ipAddress | Network address, e.g. 192.168.0.68 |
| password  | Password of the air-Q device       |

The Thing provides the following properties:
| Parameter              | Description                   |
|------------------------|-------------------------------|
| id                     | Device ID                     |
| hardwareVersion        | Hardware version              |
| softwareVersion        | Firmware version              |
| sensorList             | Available sensors             |
| sensorInfo             | Information about the sensors |
| industryVersion        | Industry version              |

## Channels

The air-Q Thing offers access to all sensor data of the air-Q, according to its version.
This includes also the Maximum Error per sensor value.
For the Maximum Error channels just add `_maxerr` to the channel names.

> | channel                   | type     | description                                                         |

Reviewer: You could make clear if a channel is read-only or writeable.

|---------------------------|----------|---------------------------------------------------------------------|

> | status                    | String   | Status of the sensors                                               |

Reviewer: Can you mention the valid states?

> | avgFineDustSize           | Number   | Average size of Fine Dust [experimental]                            |

Reviewer: You could use Units of Measure. For Example `Number:length`.

> | fineDustCnt00_3           | Number   | Fine Dust >0,3 &mu;m                                                |

Reviewer: As this is Unicode, you could use the actual character.
```suggestion
| fineDustCnt00_3           | Number:Length   | Fine Dust >0,3 µm                                                |
```

| fineDustCnt00_5           | Number   | Fine Dust >0,5 &mu;m                                                |
| fineDustCnt01             | Number   | Fine Dust >1 &mu;m                                                  |
| fineDustCnt02_5           | Number   | Fine Dust >2,5 &mu;m                                                |
| fineDustCnt05             | Number   | Fine Dust >5 &mu;m                                                  |
| fineDustCnt10             | Number   | Fine Dust >10 &mu;m                                                 |
| co                        | Number   | CO concentration                                                    |

> | co2                       | Number   | CO<sub>2</sub> concentration                                        |

Reviewer: ```suggestion
| co2                       | Number:Dimensionless   | CO<sub>2</sub> concentration                                        |
```

| dCO2dt                    | Number   | Change of CO<sub>2</sub> concentration                              |
| dHdt                      | Number   | Change of Humidity                                                  |

> | dewpt                     | Number   | Dew Point                                                           |

Reviewer: And so on
```suggestion
| dewpt                     | Number:Temperature   | Dew Point                                                           |
```

> | doorEvent                 | Switch   | Door Event (experimental)                                           |

Reviewer: Does the normal user know what is meant by this?

| temperature               | Number   | Temperature                                                         |
| timestamp                 | Time     | Timestamp of measurement                                            |
| voc                       | Number   | VOC concentration                                                   |
| uptime                    | Number   | uptime in seconds                                                   |
| wifi                      | Switch   | WLAN on or off                                                      |
| SSID                      | String   | WLAN SSID                                                           |
| password                  | String   | Device Password                                                     |
| wifiInfo                  | Switch   | Show WLAN status with LED                                           |
| timeServer                | String   | Name of Timeserver address                                          |
| location                  | Location | Location of air-Q device                                            |

> | nightmode_StartDay        | String   | Time to start day operation                                         |

Reviewer: Try to use a consistent naming scheme. Please check all.
```suggestion
| nightmodeStartDay        | String   | Time to start day operation                                         |
```

| nightmode_StartNight      | String   | End of day operation                                                |
| nightmode_BrightnessDay   | Number   | Brightness of LED during the day                                    |
| nightmode_BrightnessNight | Number   | Brightness of LED at night                                          |
| nightmode_FanNightOff     | Switch   | Switch off fan at night                                             |
| nightmode_WifiNightOff    | Switch   | Switch off WLAN at night                                            |

> | devicename                | String   | Device Name                                                         |

Reviewer: ```suggestion
| deviceName                | String   | Device Name                                                         |
```

| status                    | String               |    | Status of the sensors (usually "OK")                                |
| avgFineDustSize           | Number:Length        |    | Average size of Fine Dust [experimental]                            |
| fineDustCnt00_3           | Number:Dimensionless |    | Fine Dust >0,3 µm                                                   |
| fineDustCnt00_5           | Number:Dimensionless |    | Fine Dust >0,5 µm                                                   |
| fineDustCnt01             | Number:Dimensionless |    | Fine Dust >1 µm                                                     |
| fineDustCnt02_5           | Number:Dimensionless |    | Fine Dust >2,5 µm                                                   |
| fineDustCnt05             | Number:Dimensionless |    | Fine Dust >5 µm                                                     |
| fineDustCnt10             | Number:Dimensionless |    | Fine Dust >10 µm                                                    |
| co                        | Number:Dimensionless |    | CO concentration                                                    |

> | co2                       | Number               |    | CO₂ concentration                                                   |

Reviewer: Looks like you need to update some types here after making them `QuantityType`.

| humidityRelative          | Number:Dimensionless |    | Humidity in percent                                                 |
| humidityAbsolute          | Number               |    | Absolute Humidity                                                   |
| measureTime               | Number               |    | Milliseconds needed for measurement                                 |
| no2                       | Number               |    | NO₂ concentration                                                   |
| o3                        | Number               |    | Ozone (O₃) concentration                                            |
| o2                        | Number:Dimensionless |    | Oxygen (O₂) concentration                                           |
| performance               | Number:Dimensionless |    | Performance Index (0 to 1000)                                       |
| fineDustConc01            | Number               |    | Fine Dust concentration >1 µm                                       |
| fineDustConc02_5          | Number               |    | Fine Dust concentration >2.5 µm                                     |

> | fineDustConc10            | Number               |    | Fine Dust concentration >10 µm             fni                      |

Reviewer: ```suggestion
| fineDustConc10            | Number               |    | Fine Dust concentration >10 µm                                      |
```

Author: Is this a typo?

Reviewer: I don't know. What is the purpose of the "fni"?

Author: Sorry, I had understood that you inserted it. But actually it was _my_ typo, and I removed it now.

img[src*="#small"] {
    width:400px;
    height:324px;

> }

Reviewer: Are you sure this will work? The GitHub preview renders this as oridnary text.

Author: Right. I resized the image so it does not need a size modifier anymore.

# air-Q Binding

The air-Q Binding integrates the air analyzer [air-Q](http://www.air-q.com) device into the openHAB system.

With the binding, it is possible to subscribe to all data delivered by the air-Q device.

![air-Q image](doc/image_air-Q.png#small)

## Supported Things

> Only one Thing is supported: The `air-Q` device.

Reviewer: Better use the Thing Type ID here.
```suggestion
Only one Thing is supported: The `airq` device.
```

Author: Actually I tried to use airq when it was about the binding (because the binding name does not allow the hyphen), and air-Q when it is about the device. OK if I leave it like this?

Reviewer: I meant only this occurrence as users who use textual configuration need it, if that answers you question.

Reviewer: Can you comment on this?

This Binding was tested with an `air-Q Pro` device with 14 sensors. It also works with an `air-Q` device with 11 sensors.

## Discovery

Auto-discovery is not supported.

## Thing Configuration

The air-Q Thing must be configured with (both mandatory):

> | Parameter | Description                        |

Reviewer: There must be an empty line before and after tables.

# air-Q Binding

The air-Q Binding integrates the air analyzer <a href="http://www.air-q.com">air-Q</a> device into the openHAB system.

With the binding, it is possible to subscribe to all data delivered by the air-Q device.

> <img src="https://uploads-ssl.webflow.com/5bd9feee2fb42232fe1d0196/5e4a8dc0e322ca33891b51e4_air-Q%20frontal-p-800.png" alt="air-Q image" width="400px" height="324px" />

Author: Yes, I have received permission for air-Q to use that link for the photo (by e-mail). I will ask again to be sure I can use a local copy of it. I put it in the resources folder.

> <img src="https://uploads-ssl.webflow.com/5bd9feee2fb42232fe1d0196/5e4a8dc0e322ca33891b51e4_air-Q%20frontal-p-800.png" alt="air-Q image" width="400px" height="324px" />

Author: Permission received from the CEO of air-Q to use a local copy of the image.

## Supported Things

Only one Thing is supported: The `air-Q` device.
This Binding was tested with an `air-Q Pro` device with 14 sensors. It also works with an `air-Q` device with 11 sensors.

## Discovery

Auto-discovery is not possible in this version.

> Since the binding has to be configured at least with the password of the device, auto-discovery would be of limited value anyway.

Reviewer 2: Suggest removing this sentence and just simply state "Auto-discovery is not yet supported".

| Parameter | Description                        |
|-----------|------------------------------------|
| ipAddress | Network address, e.g. 192.168.0.68 |
| password  | Password of the air-Q device       |

The Thing provides the following properties:
| Parameter              | Description                   |
|------------------------|-------------------------------|
| id                     | Device ID                     |
| hardwareVersion        | Hardware version              |
| softwareVersion        | Firmware version              |
| sensorList             | Available sensors             |
| sensorInfo             | Information about the sensors |
| industryVersion        | Industry version              |

## Channels

The air-Q Thing offers access to all sensor data of the air-Q, according to its version.
This includes also the Maximum Error per sensor value.
For the Maximum Error channels just add `_maxerr` to the channel names.

| channel                   | type     | description                                                         |
|---------------------------|----------|---------------------------------------------------------------------|

> | Status                    | String   | Status of the sensors                                               |

Reviewer 2: ```suggestion
| status                    | String   | Status of the sensors                                               |
```

| fineDustCnt00_5           | Number   | Fine Dust >0,5 &mu;m                                                |
| fineDustCnt01             | Number   | Fine Dust >1 &mu;m                                                  |
| fineDustCnt02_5           | Number   | Fine Dust >2,5 &mu;m                                                |
| fineDustCnt05             | Number   | Fine Dust >5 &mu;m                                                  |
| fineDustCnt10             | Number   | Fine Dust >10 &mu;m                                                 |
| co                        | Number   | CO concentration                                                    |
| co2                       | Number   | CO<sub>2</sub> concentration                                        |
| dCO2dt                    | Number   | Change of CO<sub>2</sub> concentration                              |
| dHdt                      | Number   | Change of Humidity                                                  |
| dewpt                     | Number   | Dew Point                                                           |

> | door_event                | Switch   | Door Event (experimental)                                           |

Reviewer 2: ```suggestion
| doorEvent                | Switch   | Door Event (experimental)                                           |
```

Reviewer 2: This has not been changed in the XML either.

| health                    | Number   | Health Index                                                        |
| humidity                  | Number   | Humidity in percent                                                 |

> | humidity_abs              | Number   | Absolute Humidity                                                   |

Reviewer 2: humidityAbsolute or asbsoluteHumidity. This has not been changed in the XML either.

> | measuretime               | Number   | Milliseconds needed for measurement                                 |

Reviewer 2: measureTime

# airq Binding

The airq Binding integrates the air analyzer <a href="http://www.air-q.com">air-Q</a> device into the openHAB system.

With the binding, it is possible to subscribe to all data delivered by the air-Q device.

<img src="https://uploads-ssl.webflow.com/5bd9feee2fb42232fe1d0196/5e4a8dc0e322ca33891b51e4_air-Q%20frontal-p-800.png" alt="air-Q image" width="400px" height="324px" />

## Supported Things

> One only Thing is supported: the air-Q device.

Author: Why the `air-Q`?

> One only Thing is supported: the air-Q device.

Reviewer 2: You used air-Q as the label for the thing definition, so this is what will show up in the main UI when adding the device. Not sure I understand what you are asking here.

> One only Thing is supported: the air-Q device.

Author: I mean the inverted comma, quote or however you call it, what surrounds air-Q in your suggestion. I can't write it in the text field because it starts a code section. You mean to use the quote around `air-Q` only when the device is meant, or whenever I use that name in the README.md?

> One only Thing is supported: the air-Q device.

Reviewer 2: I was using it like italics, underlining the word. But that is not necessary, and if you dont like the use of it, just leave the quote out.

This Binding was tested with an air-Q Pro device with 14 sensors. It should also work with an air-Q device with 11 sensors, but it was not tested yet.

## Discovery

> Auto-discovery is not possible in this version. Since the binding has to be configured at least with the password of the device, auto-discovery would be of limited value anyway.

Reviewer 2: It is fine to not implement discovery, however is it possible to add discovery and add the device without the password? It should then just go OFFLINE and have a status that the password is wrong/missing. The user then just edits the thing to have a password and it can then go ONLINE.

Author: The implementation of the discovery on the air-Q is something odd, it does not answer to multi- or broadcasts. At this moment, I would like to publish the binding without the discovery and implement it later, once I get the appropriate information from the air-Q manufacturers. So at the moment it does not work without giving the IP address to the thing.

## Binding Configuration

The binding does not need to be configured.

## Thing Configuration

The air-Q thing must be configured with (both mandatory):
| Parameter | Description                        |
|-----------|------------------------------------|
| ipAddress | Network address, e.g. 192.168.0.68 |
| password  | Password of the air-Q device       |

The Thing provides the following properties:
| Parameter              | Description                   |
|------------------------|-------------------------------|
| id                     | Device ID                     |
| air-Q-Hardware-Version | Hardware version              |
| air-Q-Software-Version | Firmware version              |
| sensors                | Available sensors             |
| SensorInfo             | Information about the sensors |

> | Industry               | Industry version              |

Reviewer 2: Parameters should be camelCase, so consider changing them to something like this:

SensorInfo to sensorInfo
Industry  to industryVersion
air-Q-Hardware-Version to hardwareVersion

## Channels

The air-Q Thing offers access to all sensor data of the air-Q, according to its version. This includes also the Maximum Error per sensor value.

| channel                   | type     | description                                                         |
|---------------------------|----------|---------------------------------------------------------------------|

> | DeviceID                  | String   | Individual ID of the device                                         |

Reviewer 2: All channels should be camelCase. Consider changing to deviceID and Status to just `status`. etc...
